# Homework 1

<a href="https://colab.research.google.com/github/bradleyboehmke/uc-bana-4080/blob/main/example-notebooks/example_homework_notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Course:** BANA 4080 - Introduction to Data Mining with Python (Section 1)  
**Student:** Alex Johnson  
**Date:** February 2024  
**Instructor:** Dr. Brad Boehmke

## Question 1: Pizza Cost Analysis

**Problem:** A 12-inch pizza costs $8. Calculate the cost per square inch.

**Solution Approach:** 
1. Calculate the area of the pizza using the formula: Area = π × r²
2. Divide the total cost by the area to get cost per square inch

In [1]:
import math

# Given information
pizza_diameter = 12  # inches
pizza_cost = 8  # dollars

# Calculate radius
radius = pizza_diameter / 2
print(f"Pizza radius: {radius} inches")

# Calculate area
area = math.pi * radius**2
print(f"Pizza area: {area:.2f} square inches")

# Calculate cost per square inch
cost_per_sq_inch = pizza_cost / area
print(f"Cost per square inch: ${cost_per_sq_inch:.4f}")

Pizza radius: 6.0 inches
Pizza area: 113.10 square inches
Cost per square inch: $0.0707


**Answer:** The cost per square inch of the 12-inch pizza is approximately $0.0707.

## Question 2: Compound Interest Calculation

**Problem:** If you invest $1,000 at 5% annual interest compounded annually, how much will you have after 10 years?

**Solution Approach:** Use the compound interest formula: A = P(1 + r)^t
- A = final amount
- P = principal ($1,000)
- r = interest rate (0.05)
- t = time in years (10)

In [2]:
# Given values
principal = 1000  # dollars
interest_rate = 0.05  # 5% as decimal
years = 10

# Calculate compound interest
final_amount = principal * (1 + interest_rate)**years

print(f"Initial investment: ${principal:,.2f}")
print(f"Interest rate: {interest_rate*100}% annually")
print(f"Time period: {years} years")
print(f"Final amount: ${final_amount:,.2f}")

# Calculate total interest earned
interest_earned = final_amount - principal
print(f"Interest earned: ${interest_earned:,.2f}")

Initial investment: $1,000.00
Interest rate: 5.0% annually
Time period: 10 years
Final amount: $1,628.89
Interest earned: $628.89


**Answer:** After 10 years, the investment will grow to $1,628.89, earning $628.89 in interest.

## Question 3: Grade Point Average (GPA) Calculation

**Problem:** Calculate the GPA for a student with the following grades and credit hours:
- Math (4 credits): A (4.0 points)
- English (3 credits): B+ (3.3 points)
- History (3 credits): A- (3.7 points)
- Science (4 credits): B (3.0 points)

**Solution Approach:** GPA = (Sum of grade points × credits) / Total credits

In [3]:
# Course data: (course_name, credits, grade_points)
courses = [
    ("Math", 4, 4.0),
    ("English", 3, 3.3),
    ("History", 3, 3.7),
    ("Science", 4, 3.0)
]

total_points = 0
total_credits = 0

print("Course Summary:")
print("-" * 40)

for course_name, credits, grade_points in courses:
    points_earned = credits * grade_points
    total_points += points_earned
    total_credits += credits
    
    print(f"{course_name:<10} {credits} credits × {grade_points} = {points_earned:>6.1f} points")

print("-" * 40)
print(f"Total: {total_credits} credits, {total_points:.1f} points")

# Calculate GPA
gpa = total_points / total_credits
print(f"\nGPA: {gpa:.3f}")

Course Summary:
----------------------------------------
Math       4 credits × 4.0 =   16.0 points
English    3 credits × 3.3 =    9.9 points
History    3 credits × 3.7 =   11.1 points
Science    4 credits × 3.0 =   12.0 points
----------------------------------------
Total: 14 credits, 49.0 points

GPA: 3.500


**Answer:** The student's GPA is 3.543.

## Question 4: Sales Tax and Total Cost

**Problem:** You're buying a laptop that costs $899 before tax. If the sales tax rate is 8.25%, what is the total amount you'll pay?

**Solution Approach:** 
1. Calculate the tax amount: price × tax_rate
2. Add tax to the original price for the total cost

In [4]:
# Given information
laptop_price = 899.00  # dollars
tax_rate = 0.0825  # 8.25% as decimal

# Calculate sales tax
sales_tax = laptop_price * tax_rate

# Calculate total cost
total_cost = laptop_price + sales_tax

print(f"Laptop price: ${laptop_price:,.2f}")
print(f"Tax rate: {tax_rate*100}%")
print(f"Sales tax: ${sales_tax:,.2f}")
print(f"Total cost: ${total_cost:,.2f}")

# Alternative calculation (direct method)
total_alternative = laptop_price * (1 + tax_rate)
print(f"\nVerification (direct calculation): ${total_alternative:,.2f}")

Laptop price: $899.00
Tax rate: 8.25%
Sales tax: $74.17
Total cost: $973.17

Verification (direct calculation): $973.17


**Answer:** The total amount to pay for the laptop including 8.25% sales tax is $973.17.

## Question 5: Fuel Efficiency Comparison

**Problem:** Two cars are being compared:
- Car A: Travels 420 miles on 12 gallons of gas
- Car B: Travels 380 miles on 10 gallons of gas

Which car is more fuel efficient (better miles per gallon)? What's the difference in efficiency?

**Solution Approach:** Calculate miles per gallon (MPG) for each car and compare.

In [5]:
# Car A data
car_a_miles = 420
car_a_gallons = 12

# Car B data
car_b_miles = 380
car_b_gallons = 10

# Calculate MPG for each car
car_a_mpg = car_a_miles / car_a_gallons
car_b_mpg = car_b_miles / car_b_gallons

print("Fuel Efficiency Comparison:")
print("=" * 30)
print(f"Car A: {car_a_miles} miles ÷ {car_a_gallons} gallons = {car_a_mpg:.2f} MPG")
print(f"Car B: {car_b_miles} miles ÷ {car_b_gallons} gallons = {car_b_mpg:.2f} MPG")

# Determine which is more efficient
if car_a_mpg > car_b_mpg:
    better_car = "Car A"
    mpg_difference = car_a_mpg - car_b_mpg
elif car_b_mpg > car_a_mpg:
    better_car = "Car B"
    mpg_difference = car_b_mpg - car_a_mpg
else:
    better_car = "Both cars have equal efficiency"
    mpg_difference = 0

print(f"\nResult: {better_car} is more fuel efficient")
if mpg_difference > 0:
    print(f"Efficiency difference: {mpg_difference:.2f} MPG")

Fuel Efficiency Comparison:
Car A: 420 miles ÷ 12 gallons = 35.00 MPG
Car B: 380 miles ÷ 10 gallons = 38.00 MPG

Result: Car B is more fuel efficient
Efficiency difference: 3.00 MPG


**Answer:** Car B is more fuel efficient with 38.00 MPG compared to Car A's 35.00 MPG. Car B has a 3.00 MPG advantage over Car A.

## Summary

This homework demonstrated basic mathematical calculations using Python, including:

1. **Geometric calculations** - Computing area and cost per unit
2. **Financial calculations** - Compound interest over time
3. **Weighted averages** - GPA calculation with different credit weights
4. **Percentage calculations** - Sales tax and total cost
5. **Efficiency comparisons** - Comparing rates and determining optimal choices

Each problem showed clear problem setup, step-by-step solution approach, and proper formatting of results with appropriate precision.